pkg/webui/console/containers/lora-cloud-gls-form/index.js

```diff
@@ -49,21 +51,50 @@ const m = defineMessages({
   setLoRaCloudToken: 'Set LoRa Cloud token',
   deleteWarning:
     'Are you sure you want to delete the LoRaCloud Geolocation token? This action cannot be undone.',
+  queryType: 'Query type',
+  queryTypeDescription: 'What kind of geolocation query should be used',
+  multiFrame: 'Multiframe',
+  multiFrameDescription: 'Enable multiframe lookups',
+  multiFrameWindowSize: 'Multiframe window size',
+  multiFrameWindowSizeDescription:
+    'How many historical message to send as part of the request. A window size of 0 automatically determines this based on the first byte of the payload.',
+  multiFrameTimeWindow: 'Multiframe time window',
+  multiFrameTimeWindowDescription: 'How recent the historical messages should be, in minutes.',
 })
 
+const LORACLOUD_GLS_QUERY_LABELS = Object.freeze([{ value: 'TDOARSSI', label: 'LoRa® TOA/RSSI' }])
+const LORACLOUD_GLS_QUERY_TYPES = Object.freeze({
+  TDOARSSI: 'TDOARSSI',
+})
+const LORACLOUD_GLS_QUERY_VALUES = Object.freeze(Object.values(LORACLOUD_GLS_QUERY_TYPES))
+
 const validationSchema = Yup.object()
   .shape({
     data: Yup.object().shape({
       token: Yup.string().required(sharedMessages.validateRequired),
-      query: Yup.string().oneOf(['TOARSSI']),
+      query: Yup.string()
+        .oneOf(LORACLOUD_GLS_QUERY_VALUES)
+        .required(sharedMessages.validateRequired),
+      multi_frame: Yup.boolean().required(sharedMessages.validateRequired),
+      multi_frame_window_size: Yup.number()
```

Comment:
As far as I understand `multi_frame_window_size` and `multi_frame_window_age` should be set only when `multi_frame` is `true`. If thats the case, please strip these values within the validation schema.

```js
multi_frame_window_size: Yup.number().when('multi_frame', {
  is: true,
  then: schema => schema
          .min(0, Yup.passValues(sharedMessages.validateNumberGte))
          .max(16, Yup.passValues(sharedMessages.validateNumberLte))
          .required(sharedMessages.validateRequired),
  otherwise: schema => schema.strip()
})
```